MetaMask/metamask-mobile
# feat(ramp): add initial sell navigation

## **Description**

This PR refactors and adapts the navigation and hooks from the Ramp flow to support a new type of ramp: off-ramp. This is done by setting a `rampType` property in the current flow. This property will also be used in the hooks that get the data from the ramp SDK to decide wether call an on-ramp method or off-ramp method.

This PR will allow the user to open the sell flow, however **the UI elements will not reflect this**, the entities loaded (supported regions, payment methods, crypto currencies, fiat currencies, etc) will be the corresponding items to the sell flow.



## **Manual testing steps**

1. Go through the buy flow as usual by clicking on Buy buttons. The flow should not change from current production version.
2. As seen on the video, click in the action button in the middle of the bottom section
3. Click on Sell to open the Sell flow
4. Follow the flow up to the redirection from the quote to the provider's website

## **Screenshots/Recordings**


### **Before**

_n/a_

### **After**


https://github.com/MetaMask/metamask-mobile/assets/1024246/c2880178-5e82-4646-b77a-1c318a3b3819



## **Related issues**

_Fixes #???_

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
